jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,913 @@
+    elif instrument.lower() == 'miri':
+        instrument = 'MIRI'
+        dark_template = ['MIR_DARKALL', 'MIR_DARKIMG', 'MIR_DARKMRS']
+
+    # monitor_mast.instrument_inventory does not allow list inputs to
+    # the added_filters input (or at least if you do provide a list, then
+    # it becomes a nested list when it sends the query to MAST. The
+    # nested list is subsequently ignored by MAST.)
+    # So query once for each dark template, and combine outputs into a
+    # single list.
+    query_results = []
+    for template_name in dark_template:
+        # Create dictionary of parameters to add
+        parameters = {"date_obs_mjd": {"min": start_date, "max": end_date},
+                      "apername": aperture, "exp_type": template_name}
+
+        query = monitor_mast.instrument_inventory(instrument, dataproduct=JWST_DATAPRODUCTS,
+                                                  add_filters=parameters, return_data=True, caom=False)
+        if len(query['data']) > 0:
+            query_results.extend(query['data'])
+    return query_results
+
+
+class Dark():
+    def __init__(self, testing=False):
+        """
+        Loop over instrument/aperture combinations and find the number of
+        new dark current files available. If there are enough, copy the
+        files over to a working directory and run the monitor.
+
+        Parameters
+        ----------
+        testing : bool
+            For pytest. If True, an instance of Dark is created, but no
+            other code is executed.
+        """
+        # Begin logging
+        #logging.info("Beginning dark monitor")
+        apertures_to_skip = ['NRCALL_FULL', 'NRCAS_FULL', 'NRCBS_FULL']
+
+        if not testing:
+            # Get the output directory
+            self.output_dir = os.path.join(get_config()['outputs'], 'monitor_darks')
+
+            # Read in config file that defines the thresholds for the number
+            # of dark files that must be present in order for the monitor to
+            # run
+            limits = ascii.read(THRESHOLDS_FILE)
+
+            # Use the current time as the end time for MAST query
+            self.query_end = Time.now().mjd
+
+            # Loop over all instruments
+            for instrument in JWST_INSTRUMENT_NAMES:
+                self.instrument = instrument
+
+                # Identify which database tables to use
+                self.identify_tables()
+
+                # Get a list of all possible apertures from pysiaf
+                possible_apertures = list(Siaf(instrument).apernames)
+                possible_apertures = [ap for ap in possible_apertures if ap not in apertures_to_skip]
+
+                for aperture in possible_apertures:
+                    # Find the appropriate threshold for the number of
+                    # new files needed
+                    match = aperture == limits['Aperture']
+                    new_dark_threshold = limits['Threshold'][match]
```

Comment:
@bhilbert4 This variable name confused me a bit, I was thinking it had to do with a threshold applied to pixel values.  Perhaps something like `num_files_threshold` would be better here.

Reply:
Changed to `file_count_threshold`